### ISSUE TITLE
dmenu-extended doesn't honor Terminal=true in .desktop files?

### ISSUE THREAD

**Issue by xalexalex** [markjones112358/dmenu-extended#45]
Hello,
I have run into an issue.

Take the following .desktop file as an example
```
[Desktop Entry]
Type=Application
Name=ranger
Terminal=true
Exec=ranger
MimeType=inode/directory;
```
In a terminal, `xdg-open ~` correctly opens ranger in my home directory.
If I run dmenu-extended from terminal (`dmenu_extended_run`) and pick a directory (say `~/Downloads`), it is correctly opened in ranger _in the same terminal where I ran `dmenu_extended_run`_.
If i run dmenu-extended through an hotkey in i3, then this kind of programs that should run in terminal actually run in my login shell outside the X session, and I have to exit i3 to see them.

A small fix, however, makes it work as expected:
```
(In the desktop file)
Exec=urxvt -e ranger
```

Now, obviously, this isn't the correct way to fix it.

I have tested everything with `vim` instead of `ranger`, text files instead of directories, and it's the same.

EDIT: I am beginning to think that maybe the issue is not in dmenu-extended, but in xdg-open. Any thoughts?

**Comment by markjones112358** [comment]
That's interesting.
What do you have in your preferences file for:

        "terminal": "gnome-terminal", 

When you run `ranger` as per the example `.desktop` file, it should execute the script `~/.dmenuEextended_shellCommand.sh` which contains

    #! /bin/bash
    ranger

using the command `gnome-terminal -e /home/user/.dmenuEextended_shellCommand.sh`

What flavour of Linux are you running?

**Comment by xalexalex** [comment]
Regarding your questions:
In my dmenu-extended preferences I have `"terminal": "urxvt"`. I just tried with `konsole` and the issue persists.
I am running Arch Linux.

When in dmenu-extended I pick `ranger` directly, everything works fine and I get my terminal of choice running ranger.
If instead I pick a directory (which should be opened by ranger), the strange behavior appears. It seems to be mediated by `xdg-open`. It's as if it does open it in a terminal, but in the wrong one (the login terminal). The expected behavior would be to open a new terminal (under the window manager) in which to run ranger.

**Comment by markjones112358** [comment]
I'm about 50% sure I understand what you're saying.
I can select a folder from the menu and it opens in Nautilus fine, but that's the behaviour I expect.
Have you configured your system such that `xdg-open` would open a folder in `ranger`?
If so, what happens when you type a command such as

    xdg-open /home/user/Downloads

into a terminal and run that. Does it do as you expect?
I can't speak for why the terminal programs you're running are launched in the login terminal, that's bizzare.
Try setting this in your preferences file and see if this does what you want:

        "filebrowser": "gnome-terminal -e ranger", 

Then open a directory from within dmenu-extended.

**Comment by xalexalex** [comment]
I'm sorry, I had noticed my message was convoluted and I rewrote it twice, but to no avail apparently.

`xdg-open /path/to/some/folder` works fine in terminal

Setting `"filebrowser": "urxvt -e ranger"` does indeed work and make everything behave as expected.

There may be a bigger issue somewhere down below, I'm not sure it's in dmenu-extended. I'll do some digging and report back to you if I find something.

This issue is solved. Thank you!
